### PR TITLE
Upgrade qiskit_sphinx_theme to 1.12 and remove unnecessary CSS files

### PR DESCRIPTION
**Pull request by Eric-Arellano**
The 1.12 theme release is focused on stability, including CSS fixes. See https://github.com/Qiskit/qiskit_sphinx_theme/releases/tag/1.12.0rc1 for the changelog.

This PR also:

* Allows admins to trigger docs deploys via the GitHub UI.
* Enables `-j auto` with Sphinx to have parallelism. This should result in a faster docs build.
* Adds `tox -e docs-clean`

Finally, it removes these unused CSS files:

* `gallery.css` is for `sphinx-gallery`, which is not used by this project.
* `style.css` is leftover from when we used the Read The Docs theme. It's no longer relevant to qiskit_sphinx_theme.
* `custom.css` only has rules for `.toggle`, which does not exist anymore.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -41,11 +41,9 @@
               'jupyter_sphinx',
               'reno.sphinxext',
               'sphinx.ext.intersphinx',
-              "sphinxcontrib.jquery",  # Remove when changing html_theme to qiskit_ecosystem.
+              "qiskit_sphinx_theme",
```
Comment (Eric-Arellano) [inline docs/conf.py:44]: This will activate `sphinxcontrib.jquery`.

File: pyproject.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools", "wheel"]
```
Comment (Eric-Arellano) [inline pyproject.toml:2]: Setuptools' docs used to say this was necessary, but they clarified it was a bad recommendation.

**Review by Eric-Arellano — COMMENTED** [review]
Bump. cc @1ucian0

**Comment by Eric-Arellano** [comment]
Bump

**Comment by mtreinish** [comment]
Sorry this fell through the cracks, I think that this is superseded by therecently merged #32 so I'm going to close this. But, please feel free to reopen this if I'm incorrect or missing something

**Pull request closed without merging**